llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -21427,6 +21432,23 @@ bool RISCVTargetLowering::preferScalarizeSplat(SDNode *N) const {
   return true;
 }
 
+int RISCVTargetLowering::getMachineRoundingMode(RoundingMode RM) const {
+  switch (RM) {
+  case RoundingMode::TowardZero:
+    return RISCVFPRndMode::RTZ;
```

Comment:
Instead of introducing this hook, can these just be handled in selection values? It's only an encoding value question

Reply:
It now is implemented using special node for RISCV, which differ from the target-neutral FADD_ROUND an other by using target dependent encoding. Selection DAG first builds target-neutral nodes, then they are loweded to RISCV-specific nodes, which in turn are lowered by TableGen patterns.